src/Nimiq/BinaryCoding.h

```diff
@@ -0,0 +1,70 @@
+// Copyright © 2017-2019 Trust.
+//
+// This file is part of Trust. The full Trust copyright notice, including
+// terms governing use, modification, and redistribution, is contained in the
+// file LICENSE at the root of the source code distribution tree.
+
+#pragma once
+
+#include <stddef.h>
+#include <stdint.h>
+#include <vector>
+
+#define CAST32(x)    ((uint32_t)(x))
+#define CAST64(x)    ((uint64_t)(x))
+
+namespace TW {
+namespace Nimiq {
+
+/// Encodes a 16-bit value into the provided buffer.
+static inline void encode16(uint16_t val, std::vector<uint8_t>& data) {
+    data.push_back(static_cast<uint8_t>(val >> 8));
+    data.push_back(static_cast<uint8_t>(val));
+}
+
+/// Decodes a 16-bit value from the provided buffer.
+static inline uint16_t decode16(const uint8_t *_Nonnull src) {
+    return (uint16_t)(src[1]) | ((uint16_t)(src[0]) << 8);
+}
+
+/// Encodes a 32-bit value into the provided buffer.
+static inline void encode32(uint32_t val, std::vector<uint8_t>& data) {
```

Comment:
We will need to use this in other places, let's refactor into a common BinaryCoding in the root `src` folder.

Reply:
I propose `BinaryCodingLE.h` and `BinaryCodingBE.h`.
Nimiq operates on Big Endian, Bitcoin on Little Endian.

Reply:
@alejandro-isaza Would you prefer
  - a shared `BinaryCoding.h` with `encode<Size><LE/BE>()` functions
  - or two `BinaryCoding<LE/BH>.h` with colliding `encode<Size>()` functions

Reply:
Let's do option 1. A single `BinaryCoding.h` file with `encode32LE` and `encode32BE` functions.

Reply:
@alejandro-isaza Done in #83